spec/components/timeline/Timeline.spec.tsx

```diff
@@ -0,0 +1,168 @@
+import * as React from 'react';
+import { render } from '@testing-library/react';
+import { renderHook, act } from '@testing-library/react-hooks'
+
+import { useTimeline, TimeLineItem } from '../../../src/components/timeline';
+import { Badge } from '../../../src/components/badge';
+import Icon from '../../../src/components/icon/Icon';
+
+type MyItem = {
+    title: string;
+    name: string;
+    price: number;
+  };
+  
+  
+const DATA: TimeLineItem<MyItem>[] = [
+  {
+    icon: <Icon iconName="plus" />,
+    time: '2007-03-01T13:00:00Z',
+    value: { title: 'Title 1', name: 'Item 1', price: 23.45 },
+    hasBody: true,
+  },
+  {
+    icon: <Icon iconName="right-arrow" />,
+    time: '2007-03-01T17:00:00Z',
+    value: { title: 'Title 2', name: 'Item 2', price: 132.55 },
+    hasBody: true,
+  },
+  {
+    icon: <Icon iconName="block" />,
+    time: '2007-03-01T19:00:00Z',
+    value: { title: 'Title 3', name: 'Item 3', price: 34.12 },
+    hasBody: true,
+  },
+];
+  
+const Header = (item: MyItem) => <div>{item.title}</div>;
+const Body = (item: MyItem) => (
+  <div data-testid="body">
+    <div>
+      <Badge variant="positive" className="tk-mb-h">{item.name}</Badge>
+    </div>
+    <div>
+      <Badge variant="neutral">{item.price}</Badge>
+    </div>
+  </div>
+);
+  
+
+describe('Timeline Component', () => {
+  it('render Timeline', () => {
+    const { result } = renderHook(() => useTimeline<MyItem>({
+      items: DATA,
+      itemBodyRenderer: Body,
+      itemHeaderRenderer: Header,
+    }));
+    const { Timeline, timelineProps, isAllExpanded } = result.current;
+
+    expect(isAllExpanded).toEqual(false);
+    const {container} = render(<Timeline {...timelineProps}/>);
+    expect(container.firstChild).toHaveClass('tk-timeline');
+  });
+
+  it('isAllExpanded', () => {
```

Comment:
We should try and test what the user sees instead of some property. Can we make sure all the items are visible when isAllExpanded is set?

Reply:
I updated the test and now I verify that all items are visible